include/zephyr/net/openthread.h

```diff
@@ -73,13 +73,47 @@ struct openthread_context {
  * INTERNAL_HIDDEN @endcond
  */
 
+/**
+ * @brief OpenThread state change callback structure
+ *
+ * Used to register a callback in the callback list. As many
+ * callbacks as needed can be added as long as each of them
+ * are unique pointers of struct openthread_state_changed_cb.
+ * Beware such structure should not be allocated on stack.
+ */
+struct openthread_state_changed_cb {
```

Comment:
Not sure if it could be helpful to add a `void *user_data` to the callback struct?

We would have to define a zephyr specific callback function type, for example:
```c
typedef void (*openthread_state_changed_cb)
	(otChangedFlags flags, struct openthread_context *ot_context, void *user_data);

struct openthread_state_changed_entry {
	openthread_state_changed_cb cb;
	void *user_data;

	/**
	 * Internally used field for list handling
	 *  - user must not directly modify
	 */
	sys_snode_t node;
};
```

Reply:
I've added `void *user_data` in the callback.